chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx

```diff
@@ -160,9 +160,9 @@ export const LiveChatWidgetStateful = (props: ILiveChatWidgetProps) => {
             if (localState) {
                 // adding the reconnect logic for the case when customer tries to reconnect from a new browser or InPrivate browser
                 if (isReconnectEnabled(props.chatConfig) === true && !isPersistentEnabled(props.chatConfig)) {
-                    await handleChatReconnect(chatSDK, props, dispatch, setAdapter, initStartChat, state);
+                    const noValidReconnectId = await handleChatReconnect(chatSDK, props, dispatch, setAdapter, initStartChat, state);
                     // If chat reconnect has kicked in chat state will become Active or Reconnect. So just exit, else go next
-                    if (state.appStates.conversationState === ConversationState.Active || state.appStates.conversationState === ConversationState.ReconnectChat) {
+                    if (!noValidReconnectId && (state.appStates.conversationState === ConversationState.Active || state.appStates.conversationState === ConversationState.ReconnectChat)) {
                         return;
                     }
```

Comment:
this block seems to be the same as LoC 148 to LoC 151,  could we reuse it instead?

Reply:
refactored this